core/server/models/user.js

```diff
@@ -536,53 +541,52 @@ User = ghostBookshelf.Model.extend({
                             }
 
                             s = (remaining > 1) ? 's' : '';
-                            return Promise.reject(new errors.UnauthorizedError(i18n.t('errors.models.user.incorrectPasswordAttempts', {remaining: remaining, s: s})));
+                            return Promise.reject(new errors.UnauthorizedError({message: i18n.t('errors.models.user.incorrectPasswordAttempts', {remaining: remaining, s: s})}));
 
                             // Use comma structure, not .catch, because we don't want to catch incorrect passwords
-                        }, function handleError(error) {
+                        }, function handleError(err) {
                             // If we get a validation or other error during this save, catch it and log it, but don't
                             // cause a login error because of it. The user validation is not important here.
-                            errors.logError(
-                                error,
-                                i18n.t('errors.models.user.userUpdateError.context'),
-                                i18n.t('errors.models.user.userUpdateError.help')
-                            );
-                            return Promise.reject(new errors.UnauthorizedError(i18n.t('errors.models.user.incorrectPassword')));
+                            return Promise.reject(new errors.UnauthorizedError({
+                                err: err,
+                                context: i18n.t('errors.models.user.incorrectPassword'),
+                                help: i18n.t('errors.models.user.userUpdateError.help')
+                            }));
                         });
                     }
 
                     return Promise.resolve(user.set({status: 'active', last_login: new Date()}).save({validate: false}))
-                        .catch(function handleError(error) {
+                        .catch(function handleError(err) {
+                            logging.error(new errors.GhostError({
+                                err: err,
+                                context: i18n.t('errors.models.user.userUpdateError.context'),
+                                help: i18n.t('errors.models.user.userUpdateError.help')
+                            }));
+
                             // If we get a validation or other error during this save, catch it and log it, but don't
                             // cause a login error because of it. The user validation is not important here.
-                            errors.logError(
-                                error,
-                                i18n.t('errors.models.user.userUpdateError.context'),
-                                i18n.t('errors.models.user.userUpdateError.help')
-                            );
                             return user;
                         });
-                }, errors.logAndThrowError);
```

Comment:
Again, all these small changes matter a lot IMO? They are hidden away by all the changes to the error signature. I really don't think that this should be in the same PR.